libraries/AP_HAL_ChibiOS/hwdef/FlywooF745/README.md

```diff
@@ -0,0 +1,102 @@
+# Flywoo GOKU GN 745 40A AIO
+
+https://flywoo.net/products/goku-gn745-40a-aio-bl_32-mpu6000-25-5-x-25-5
+
+The Flywoo GOKU GN 745 AIO is a flight controller produced by [Flywoo](https://flywoo.net/).
+
+## Features
+
+- MCU: STM32F745 32-bit processor,216MHz, 512Kbytes Flash
+- IMU: MPU6000 (SPI)
+- Barometer: BMP280
+- Onboard LED：WS2812*4
+- USB VCP Driver (all UARTs usable simultaneously; USB does not take up a UART)
+- 7 hardware UARTS (UART1,2,3,4,5,6,7)
+- Supports serial receivers (SBUS, iBus, Spektrum, Crossfire) only.
+- PPM and PWM receivers are not supported.
+- Onbord 8Mbytes for Blackbox logging
+- 9V Power Out: 1.5A max
+- 5V Power Out: 2.0A max
+- 3.3V Power Out: 0.5A max
+- Dimensions: 33.5x33.5mm
+- Mounting Holes: Standard 25.5/26.5mm square to center of holes 
+- Weight: 8.5g
+
+- Built-in 40A BL_32 4in1 ESC
+- Support BLheli / BLHELI_32
+- Support PWM, Oneshot125, Oneshot42, Multishot, Dshot150, Dshot300, Dshot600,Dshot1200
+- Input Voltage: 2-6S Lipo
+- Continuous Current: 40A
+- Firmware: BLHELI_32
+
+## Pinout
+
+![GOKU GN 745 40A AIO](GOKUGN745AIO_Pinout.png "GOKU GN 745 40A AIO")
+
+## UART Mapping
+
+The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
+receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
+|Name|Pin|Function|
+|:-|:-|:-|
+|SERIAL0|COMPUTER|USB|
+|SERIAL1|RX1/TX1|UART1|
+|SERIAL2|TX2/RX2|UART2|
+|SERIAL3|TX3/RX3|UART3 (RC Input/Output)|
+|SERIAL4|TX4/RX4|UART4|
+|SERIAL5|TX5/RX5|UART5|
+|SERIAL6|TX6/RX6|UART6 (GPS)|
+|SERIAL7|TX7/RX7|UART7|
+
+UART3 supports RX and TX DMA. UART1, UART2, UART4, and UART6 supports TX DMA. UART5 and UART7 do not support DMA.
+
+## RC Input
+ 
+RC input is configured on the UART3, which supports serial RC protocols. This board does not support PPM.
+  
+## OSD Support
+
+The GOKU GN 745 AIO supports OSD using OSD_TYPE 1 (MAX7456 driver).
+
+## PWM Output
+
+The GOKU GN 745 AIO supports up to 8 PWM outputs. The pads for motor output ESC1 to ESC4 on the above diagram are the first 4 outputs, there are four additional pads for PWM 5-8. All 8 outputs support DShot as well as all PWM types.
```

Comment:
two support BDIR?

Reply:
@andyp1per does the configuration with BIDIR flags on two inputs mean it only has bidirectional d-shot on two PWM outputs, or does it mean there is bidirectional dshot on 4 channels? I remember you saying it is shared by the two channels on the same timer.

Reply:
The bidir channel is shared between outputs on the same timer pair of 1/2 or 3/4 - so you have 4 bi-dir channels here

Reply:
Thanks. Updating the Readme to reflect bidirectional dshot being available on the motor pads.